### DISCUSSION TITLE
Full Separation of Duties Structure

### DISCUSSION THREAD

**Discussion by diloreto** [apiflask/apiflask#225] [Q&A]
Thanks for creating this framework! I noticed in your tutorial that you had schema and models defined right inside the main app (where `app` is instantiated). We're building an app that will have dozens of models/schemas, hundreds of routes, and each database table (schema) will have 5-10 functions (Create/Read/Update/Delete plus several others). 

We don't want our app.py script to become littered and monolithic. Can you provide a recommendation on how you would structure your application/separate files to accommodate such a size / growth rate?

Thank you!

**Comment by greyli** [comment]
Hi, sorry for the late response. I'm so happy you like APIFlask!

Since APIFlask is built on top of Flask, you can apply any technology or best practice of project organizing of Flask. For example:

- [Application Factory](https://flask.palletsprojects.com/en/2.1.x/patterns/appfactories/)
- [Blueprint](https://flask.palletsprojects.com/en/2.1.x/blueprints/)
- [Packages](https://flask.palletsprojects.com/en/2.1.x/patterns/packages/)

On top of the application factory and blueprint pattern, for simple applications, just create models.py, and schemas.py to separate different logic:

```
app/
    - __init__.py
    - blueprints
        - __init__.py
        - users.py
        - posts.py
        - categories.py
    - schemas.py
    - models.py
```

Then you can turn them into packages and separate classes with different modules when your application grows:

```
app/
    - __init__.py
    - blueprints
        - __init__.py
        - users.py
        - posts.py
        - categories.py
    - schemas/
        - __init__.py
        - users.py
        - posts.py
        - categories.py
    - models/
        - __init__.py
        - users.py
        - posts.py
        - categories.py
```
